arcgis-ios-sdk-samples/Route and directions/Display device location with NMEA data sources/SimulatedNMEADataSource.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+protocol SimulatedNMEADataSourceDelegate: AnyObject {
+    /// A mock delegate method to receive NMEA data updates.
+    /// Replace it with the real one to receive NMEA sentences from a GPS dongle.
+    /// - Parameter nmeaData: The latest NMEA data.
+    func dataSource(didUpdate nmeaData: Data)
+}
+
+class SimulatedNMEADataSource {
+    /// The playback speed multiplier.
+    private let playbackSpeed: Double
+    /// An iterator to hold and loop through the mock NMEA data.
+    private var nmeaDataIterator: CircularIterator<Data>
+    /// A timer to periodically update NMEA data.
+    private var timer: Timer?
+    /// A delegate to notify view controllers.
+    weak var delegate: SimulatedNMEADataSourceDelegate?
+    
+    /// Load locations from NMEA sentences.
+    /// Read mock NMEA sentences line by line and group them by the timestamp.
+    /// - Parameters:
+    ///   - nmeaSourceFile: The URL of the NMEA source file.
+    ///   - speed: The playback speed multiplier.
+    init(from nmeaSourceFile: URL, speed: Double = 1.0) {
+        // An empty container for NMEA data.
+        var dataSplittedBySecond = [Data]()
```

Comment:
```suggestion
        var dataSplitBySeconds = [Data]()
```
I think this is a bit more grammatically correct. Or maybe just `dataBySeconds`?